controllers/cluster_controller.go

```diff
@@ -744,22 +710,22 @@ func getDynamicConfHash(configMap *corev1.ConfigMap, pClass fdbtypes.ProcessClas
 	return GetJSONHash(data)
 }
 
-func (r *FoundationDBClusterReconciler) getPodClient(cluster *fdbtypes.FoundationDBCluster, instance FdbInstance) (FdbPodClient, error) {
+func (r *FoundationDBClusterReconciler) getPodClient(cluster *fdbtypes.FoundationDBCluster, instance FdbInstance) (FdbPodClient, string) {
```

Comment:
This was something I deliberated, and I should have brought this up in the description. I think that the common cases where the we can't get the pod client, it's because of a temporary condition where we need to wait, and I think we want to move away from using errors to model that. However, it also feels a little awkward to propagate strings from a method like this.

Reply:
Assuming we don't care about error's that happen here we could change this in a way that `getPodClient` logs the issue (e.g. the same string that we currently put ass message) and then just always return nil or the client. The only problem here would be when a "real" error occurs. Or we return 3 values `(FdbPodClient, string, error)` that would be a compromise and we still would catch "real" errors. If that makes sense.